### PR TITLE
SAA-281 Roles

### PR DESCRIPTION
**Pull request by markford-moj**
Protected these endpoints with roles 'ACTIVITY_HUB', 'ACTIVITY_HUB_LEAD', 'ACTIVITY_ADMIN' (any one) 

- POST /activities/{activityId} – Create an activity 
- POST /schedules/{scheduleId}/allocations – Allocate offender to schedule

Protected this endpoint with the 'ACTIVITY_ADMIN' role 

- PUT /attendances – Updates attendance records

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleService.kt [hunk 1]
```diff
@@ -82,6 +83,7 @@ class ActivityScheduleService(
 
   fun getScheduleById(scheduleId: Long) = repository.findOrThrowNotFound(scheduleId).toModelSchedule()
 
+  @PreAuthorize("hasAnyRole('ACTIVITY_HUB', 'ACTIVITY_HUB_LEAD', 'ACTIVITY_ADMIN')")
   fun allocatePrisoner(scheduleId: Long, request: PrisonerAllocationRequest) {
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleService.kt:87]: Service methods or on the controller methods? I've seen it done on both, but recent services have this on the controller methods.

**Review by timharrison-moj — COMMENTED** [review]
LGTM. Just the question of whether the preauth is clearer on controller methods? Can also add comments into the swagger notes about what role is required?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleService.kt (hunk 1 above)
Comment (markford-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleService.kt:87]: Really? Ive never ever seen this done on the controller methods.

Comment (markford-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ActivityScheduleService.kt:87]: Added to the controller methods and added to the swagger description to mention the roles required. Also added the bearer token security schema to the swagger config. Build was failing due to the circle ci secrets i guess

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**